Framework/Algorithms/src/MaskBins.cpp

```diff
@@ -75,40 +61,31 @@ void MaskBins::exec() {
     throw std::invalid_argument(msg.str());
   }
 
-  //---------------------------------------------------------------------------------
-  // what spectra (workspace indices) to load. Optional.
-  this->spectra_list = this->getProperty("SpectraList");
-  if (!this->spectra_list.empty()) {
-    const int numHist = static_cast<int>(inputWS->getNumberHistograms());
-    //--- Validate spectra list ---
-    for (auto wi : this->spectra_list) {
-      if ((wi < 0) || (wi >= numHist)) {
-        std::ostringstream oss;
-        oss << "One of the workspace indices specified, " << wi
-            << " is above the number of spectra in the workspace (" << numHist
-            << ").";
-        throw std::invalid_argument(oss.str());
-      }
-    }
+  // Copy indices from legacy property
+  std::vector<int> spectraList = this->getProperty("SpectraList");
+  if (!spectraList.empty()) {
+    if (!isDefault("InputWorkspaceIndexSet"))
+      throw std::runtime_error("Cannot provide both InputWorkspaceIndexSet and "
```

Comment:
See [EnabledWhenProperty](https://github.com/mantidproject/mantid/blob/master/Framework/Kernel/src/EnabledWhenProperty.cpp) for nicer behaviour via the GUI for the algorithm.

Reply:
@OwenArnold unless you really insist on this I am happy to go ahead and merge

Reply:
No I've already approved, so go ahead.